doc/stdenv/cross-compilation.chapter.md

```diff
@@ -0,0 +1,167 @@
+| build                      | build                        |
+| build                      | host                         |
+| build                      | target                       |
+| host                       | host                         |
+| host                       | target                       |
+| target                     | target                       |
+
+
+Some examples will make this table clearer. Suppose there's some package that is being built with a `(build, host, target)` platform triple of `(foo, bar, baz)`. If it has a build-time library dependency, that would be a "host → build" dependency with a triple of `(foo, foo, *)` (the target platform is irrelevant). If it needs a compiler to be built, that would be a "build → host" dependency with a triple of `(foo, foo, *)` (the target platform is irrelevant). That compiler, would be built with another compiler, also "build → host" dependency, with a triple of `(foo, foo, foo)`.
+
+### Cross packaging cookbook {#ssec-cross-cookbook}
+
+Some frequently encountered problems when packaging for cross-compilation should be answered here. Ideally, the information above is exhaustive, so this section cannot provide any new information, but it is ludicrous and cruel to expect everyone to spend effort working through the interaction of many features just to figure out the same answer to the same common problem. Feel free to add to this list!
+
+**Q:** What if my package's build system needs to build a C program to be run under the build environment?
+```
+depsBuildBuild = [ buildPackages.stdenv.cc ];
+```
+Add it to your `mkDerivation` invocation.
+
+**Q:** My package fails to find `ar`.
+
+**A:** Many packages assume that an unprefixed `ar` is available, but Nix doesn't provide one. It only provides a prefixed one, just as it only does for all the other binutils programs. It may be necessary to patch the package to fix the build system to use a prefixed `ar`.
+
+**Q:** My package's testsuite needs to run host platform code.
+```
+doCheck = stdenv.hostPlatform == stdenv.buildPlatfrom;
+```
+Add it to your `mkDerivation` invocation.
+
+## Cross-building packages {#sec-cross-usage}
+
+Nixpkgs can be instantiated with `localSystem` alone, in which case there is no cross-compiling and everything is built by and for that system, or also with `crossSystem`, in which case packages run on the latter, but all building happens on the former. Both parameters take the same schema as the 3 (build, host, and target) platforms defined in the previous section. As mentioned above, `lib.systems.examples` has some platforms which are used as arguments for these parameters in practice. You can use them programmatically, or on the command line:
+
+```ShellSession
+$ nix-build '<nixpkgs>' --arg crossSystem '(import <nixpkgs/lib>).systems.examples.fooBarBaz' -A whatever
+```
+
+> **Note**: Eventually we would like to make these platform examples an unnecessary convenience so that
```

Comment:
This note used extend down below the shell session and paragraph.

![image](https://user-images.githubusercontent.com/4804/100965313-51396780-34df-11eb-848f-fd5d317fe8c9.png)